src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -585,4 +585,27 @@ public void ListenNamedPipe(string pipeName, Action<ListenOptions> configure)
         configure(listenOptions);
         CodeBackedListenOptions.Add(listenOptions);
     }
+
+    /// <summary>
+    /// Return true if there is a default or development server certificate.
+    /// Should not be called before the configuration is loaded, if there is one.
+    /// </summary>
+    /// <returns>True if there is a default or development server certificate, false otherwise.</returns>
+    /// <exception cref="InvalidOperationException">If there is a configuration and it has not been loaded.</exception>
+    public bool CheckDefaultCertificate()
+    {
+        if (ConfigurationLoader is { IsLoaded: false })
```

Comment:
Can this call ConfigurationLoader.Load() instead of throwing?

Reply:
@halter73 Suggested doing it this way because invoking `Load` ourselves would put the burden on us to reload on change.

Reply:
We'd have to reload if CheckDefaultCertificate were called again? That seems reasonable.

Reply:
(i.e. other  consumers of the configuration might see stale values.)

Reply:
> @halter73 Suggested doing it this way because invoking `Load` ourselves would put the burden on us to reload on change.

We should capture the reload token during the first call to `Load` regardless. If we start calling `Load` ourselves in more places, we should fix the bug at the same time. Otherwise, we risk not picking up new endpoints or certa added to Kestrel config between the manual call to `Load` and a later call to `StartAsync`. We could always call `Reload` during `StarAsync`, but it'd be more efficient to only reload if the configuration has actually changed.

That said, this is a pretty narrow edge case and an existing big.